magefiles/targets/generate_proto.go

```diff
@@ -83,6 +85,19 @@ func (Generate) ProtobufPython(ctx context.Context) error {
 	return nil
 }
 
+// Build the typescript service generator plugin
+func (Generate) TypescriptServiceGenerator() error {
```

Comment:
This should be part of the `Build` namespace, and ProtobufTypescript should depend on it via `mg.Deps(Build.TypescriptServiceGenerator)`

Reply:
Done.

Is there a better way to execute a command in a different directory?
```
        webPath := path.Join(cwd, "web")
	os.Chdir(webPath)

	runError := sh.RunV("yarn", "build:service-generator")
	os.Chdir(cwd)
```

Reply:
Yes, but not using sh.Run. You would have to use exec.Command directly.

Reply:
Updated with exec.Command. Let me know if I can improve further.